tests/test_forecast/test_forecast.py

```diff
@@ -0,0 +1,209 @@
+from __future__ import unicode_literals
+
+import boto3
+import sure  # noqa
+
+from parameterized import parameterized
+from botocore.exceptions import ClientError
+from nose.tools import assert_raises
+
+from moto import mock_sts, mock_forecast
+from moto.core import ACCOUNT_ID
+from moto.forecast.models import DatasetGroup
+
+region = "us-east-1"
+account_id = None
+valid_domains = [
+    "RETAIL",
+    "CUSTOM",
+    "INVENTORY_PLANNING",
+    "EC2_CAPACITY",
+    "WORK_FORCE",
+    "WEB_TRAFFIC",
+    "METRICS",
+]
+
+
+@parameterized(valid_domains)
+@mock_forecast
+@mock_sts
+def test_forecast_dataset_group_create(domain):
+    name = "example_dataset_group"
+    client = boto3.client("forecast", region_name=region)
+    response = client.create_dataset_group(DatasetGroupName=name, Domain=domain)
+    response["ResponseMetadata"]["HTTPStatusCode"].should.equal(200)
+    response["DatasetGroupArn"].should.equal(
+        "arn:aws:forecast:" + region + ":" + ACCOUNT_ID + ":dataset-group/" + name
+    )
+
+
+@mock_forecast
+@mock_sts
+def test_forecast_dataset_group_create_invalid_domain():
+    name = "example_dataset_group"
+    client = boto3.client("forecast", region_name=region)
+
+    with assert_raises(ClientError):
```

Comment:
Can we have some minimal assertions, that the correct error was thrown? Typically this would look like:
```
with assert_raises(ClientError) as ex:
    do_something()
ex.exception.response["Error"]["Message"].should.equal("This is what's wrong: ...")
```

Reply:
Of course. I've added assertions to all exception checking tests based on this suggestion.